### PR TITLE
Added Karatsuba Algo in Dart

### LINKED ISSUE #2248: Karatsuba Algorithm
- [ ] C - @Angshuman032001
- [x] C++
- [ ] C# - @BishalKrChanda
- [ ] CoffeeScript - @kichloo
- [x] Dart - @heisastark
- [x] Go - @HardevKhandhar
- [ ] Java - @Mustafiz04
- [x] JS - @abhu-A-J
- [ ] Kotlin - @RaghavAwasthi
- [x] Php - @JasbirCodeSpace
- [ ] Python - @lakshyajit165
- [ ] Readme - @lakshyajit165
- [ ] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript - @shivanidalmia1203

### PR DESCRIPTION
**Pull request by heisastark**
Fixes #2248

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
Added Kartasuba Algo in Dart

#### Languages Used:
Dart

#### Files Added:
Karatsuba_Algorithm.dart

Thanks!

### CODE REVIEW

File: Karatsuba_Algorithm/Karatsuba_Algorithm.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    int s2 = karatSuba(num1_L + num1_H, num2_L + num2_H);
+    int s3 = karatSuba(num1_H, num2_H);
+    int s4 = s2 - s3 - s1;
+
+    // karatsuba formula for finding product
+    int result = s3 * pow(10, 2 * half) + s4 * pow(10, half) + s1;
+
+    return result;
+}
+
```
Comment (asha15) [inline Karatsuba_Algorithm/Karatsuba_Algorithm.dart:42]: remove this line

File: Karatsuba_Algorithm/Karatsuba_Algorithm.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+void main(){
+    String a, b;
+    a = stdin.readLineSync(); 
+    b = stdin.readLineSync();
+    print(karatSuba(int.parse(a), int.parse(b)));
+}
```
Comment (asha15) [inline Karatsuba_Algorithm/Karatsuba_Algorithm.dart:49]: keep a line space after this

**Comment by heisastark** [comment]
@asha15 Ma'am, I'm unable to squash the commits as i fixed the issues directly from Github editor so no local copy is present

**Comment by asha15** [comment]
so you can clone the forked repo and squash it right?

**Comment by asha15** [comment]
@heisastark 

> git rebase -i HEAD~<number of the commits>

have you tried this command?

**Comment by heisastark** [comment]
@asha15 Yes Ma'am. Even tried to force push and soft reset but no avail. Although, force push does squash the commits but adds misc files, which happened in the former case.

**Comment by heisastark** [comment]
@asha15 Should i create a new PR?

**Pull request merged**